reliability/single-fit/trace/src/test/kotlin/com/datadog/android/trace/integration/otel/OtelTracerProviderTest.kt

```diff
@@ -803,7 +808,10 @@ internal class OtelTracerProviderTest {
         val keptSpans = spansWritten.filter {
             it.getInt(SAMPLING_PRIORITY_KEY) == PrioritySampling.USER_KEEP.toInt()
         }
-        val offset = 10
+        // Because of the way sampling works the deviation can be quite high so we will have to use an offset of 15%
+        // here to make sure this test will never be flaky
+        val offset = numberOfSpans * 15 / 100
```

Comment:
nit: I think here we should use float types, otherwise we are loosing a lot by doing an integer division.

```suggestion
        val offset = numberOfSpans * 15f / 100
```

Reply:
the offset needs to be an int in the end so I will keep it like that.